Implement image store

joieng1 opened the pull request:
So I added a backend route to upload the image to the S3 bucket and return a URL. The frontend code takes the URL returned from the upload backend route and submits that into MongoDB. The new frontend removes the image URL prompt as it is no longer needed.

joieng1 committed:
feat: created route to handle img upload to aws

joieng1 committed:
feat: edit frontend to work with new backend

ryanhu021 commented on backend/routes/prizes.ts:
```diff
@@ -32,4 +56,34 @@ router.patch("/:id", async (req: any, res: any) => {
+
+  const fileName =
+    Math.random().toString(36).substring(2, 15) +
+    Math.random().toString(36).substring(2, 15);
```
This could result in duplicate file names, which will cause the upload to fail. You could incorporate `Date.now()` to get the current date/time, which should always be unique

joieng1 committed:
fix: change image url links to use Date.now()

joieng1 committed:
feat: implemented image store in log hours

joieng1 commented:
Hi, I fixed the duplicate file name issue and implemented the image store feature onto the log hours page. When editing a shift as an admin, the admin can view the image the user uploaded if an image exists. Both editing and uploading images on log hours page works and is stored in the same bucket that stores prize images.

joieng1 committed:
fix: shift slot now shows img instead of img link

joieng1 committed:
fix: img now only displays in shift slot if exists
